wgsl/index.bs

```diff
@@ -1043,6 +1058,20 @@ array_decoration
   </xmp>
 </div>
 
+<div class='example' heading='Access qualifier'>
+  <xmp>
+    # Storage buffers
+    var<storage> buf1 : [[access(read)]] Buffer;       # Can read, cannot write.
+    var<storage> buf2 : [[access(read_write)]] Buffer; # Can both read and write
+
+    # Access qualifier attached through a type alias.
+    type ROBuffer1 = [[access(read)]] Buffer;
```

Comment:
Hmm, it's strictly speaking not necessary to support that typedef. Could we consider postponing it to after MVP?
I just don't want to complicate it here beyond necessity

Reply:
It falls out of how it's defined. Type type alias has a `type_decl` on the right. That type decl is where we're attaching the access decoration. Not having it here would be more complicated, heh.

Reply:
With this API, we have 2 ways to attach an access qualifier. It may or may not be on a type. Which means, secretly, you are introducing a pseudo-type here that can be clarified. I believe this is over-complicating things.

As an alternative, the `type_decl` in general would not have any decorations expected by the grammar. Instead, decorations would only be expected on the `var` declaration before the type.

Reply:
Interesting, so you're saying we change:

```
variable_ident_decl
  : IDENT COLON type_decl
```

Something like:
```
variable_ident_decl
  : IDENT COLON variable_type_decorations* type_decl

variable_type_decorations
  : ATTR_RIGHT (variable_type_decoration COMMA)* variable_type_decoration ATTR_LEFT
 
variable_type_decoration
 : ACCESS PAREN_LEFT access_type PAREN_RIGHT
```

to accept an access decoration after the `COLON`. This technically allows attaching the decoration to _any_ variable. We'll then need to validate that it's only attached to the correct types.

Reply:
Right. We shouldn't need `variable_type_decoration` at all, btw, given that we merged #689

Reply:
We'd still end up with:

```
variable_ident_decl
  : IDENT COLON variable_decoration* type_decl
```

which then becomes a question if it's confusing we're overloading variable_decoration that way.

Reply:
yeah. It's just that the variable declaration scope is strictly less than the type declaration scope, so reducing this decorator visibility makes a tighter API.

Reply:
done